[Docs] Index patterns REST API docs (#100549)
elastic/kibana: docs/api/index-patterns/update-fields.asciidoc

[[index-patterns-fields-api-update]]
=== Update index pattern fields API
++++
<titleabbrev>Update index pattern fields metadata</titleabbrev>
++++

> experimental[] Update fields endpoint allows you to update fields presentation metadata, such as `count`,

Reviewer: ```suggestion
experimental[] Update fields presentation metadata, such as `count`,
```

`customLabel`, and `format`. You can update multiple fields in one request. Updates

> are merged with persisted metadata. To remove existing metadata specify `null` as value.

Reviewer: ```suggestion
are merged with persisted metadata. To remove existing metadata, specify `null` as the value.
```

[[index-patterns-fields-api-update-request]]
==== Request

`POST <kibana host>:<port>/api/index_patterns/index_pattern/<id>/fields`

`POST <kibana host>:<port>/s/<space_id>/api/index_patterns/index_pattern/<id>/fields`

[[index-patterns-fields-api-update-path-params]]
==== Path parameters

`space_id`::
(Optional, string) An identifier for the space. If `space_id` is not provided in the URL, the default space is used.

`id`::

> (Required, string) The ID of index pattern fields of which to update.

Reviewer: ```suggestion
(Required, string) The ID of the index pattern fields you want to update.
```

[[index-patterns-fields-api-update-request-body]]
==== Request body

`fields`::

> (Required, object) fields update spec. See examples below.

Reviewer: ```suggestion
(Required, object) The field object.
```
